server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/InternalHistogram.java

```diff
@@ -382,7 +383,46 @@ private double round(double key) {
         return Math.floor((key - emptyBucketInfo.offset) / emptyBucketInfo.interval) * emptyBucketInfo.interval + emptyBucketInfo.offset;
     }
 
-    private void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
+    private int getTotalBucketCount() {
+        int bucketCount = 0;
+        if (emptyBucketInfo != null) {
+            double min = emptyBucketInfo.minBound;
+            double max = emptyBucketInfo.maxBound;
+            double intervalWidth = 0;
+
+            int i = 0;
+            double key = min;
+            while (key < max && i++ < 10) {
+                bucketCount++;
+                double nextKey = nextKey(key);
+                intervalWidth = Math.max(intervalWidth, nextKey - key);
+                key = nextKey;
+            }
```

Comment:
This width estimation may not be necessary for `InternalHistogram`. 

For `InternalDateHistogram`, the datetime rounding can be weird, and the first/last bucket bounds can be different.

I think here, you can just replace `intervalWidth` with `emptyBucketInfo.interval`.